### ISSUE TITLE
Sanitize Guided WP setting

### ISSUE THREAD

**Issue by khancyr** [ArduPilot/ardupilot#16483]
## Bug report

**Issue details**

On contrary to Auto mode, we don't sanitize WP on Guided mode. Therefore using something like 
```
    copter.mav.mav.set_position_target_global_int_send(
        0,  # timestamp
        copter.target_system,  # target system_id
        1,  # target component id
        mavutil.mavlink.MAV_FRAME_GLOBAL_INT,
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_VX_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_VY_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_VZ_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_FORCE_SET |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_IGNORE |
        mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE,
        0,  # lat
        0,  # lon
        0,  # alt
        0,  # vx
        0,  # vy
        0,  # vz
        0,  # afx
        0,  # afy
        0,  # afz
        0,  # yaw
        0,  # yawrate
    )
```
trigger a Panic on SITL. 

What should be the best action ? 
- Reject the WP
- Change to current location 

**Version**
Master at https://github.com/ArduPilot/ardupilot/commit/275c2849cf38d32c46e868e7f3e216c8fee5aee7, should be present on current stable release.
Plane should be less impacted as it doesn't support position_target_global_int

**Platform**
[ X ] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
_What type of airframe (flying wing, glider, hex, Y6, octa etc)_

**Hardware type**
_What autopilot hardware was used? (Pixhawk, Cube, Pixracer, Navio2, etc)_

**Logs**
_Please provide a link to any relevant logs that show the issue_

**Comment by rmackay9** [comment]
I must say that I'm not a fan of interpreting of "0" as the current lat,lon or alt although it could make sense to do that to be consistent with Auto... I guess I would vote for rejecting the command if it is invalid or too far from the current location.

Might be worth dicussing on the devcall if you're planning to add the check.

**Comment by khancyr** [comment]
Yes, I was assuming that we were rejecting (0, 0) waypoint but that isn't always the case. 
I am also in favor of answering a rejection instead of silently overriding with the current position.

I note this to not forget it and intend that we discuss it on next EU devcall as that is something important.

**Comment by peterbarker** [comment]
On Tue, 2 Feb 2021, Pierre Kancir wrote:

> 
> Yes, I was assuming that we were rejecting (0, 0) waypoint but that isn't always the case.
> I am also in favor of answering a rejection instead of silently overriding with the current position.
> 
> I note this to not forget it and intend that we discuss it on next EU devcall as that is something important.

FWIW I was looking at whether it was possible to abstract out a method to 
grab location from a command; lots of places do it ('though not so many 
_int places...
)

**Comment by hendjoshsr71** [comment]
Was this discussed on the devcall with an outcome?  

I managed to do the same thing with a 0,0,0 position during recent guided testing. 
I'm guessing rejection is probably the way forward.

**Comment by tridge** [comment]
discussed on EU dev call, decided we really need an initialised bool on Location rather than assuming 0,0,0 is uninitialised

**Comment by pshkv** [comment]
This is a compelling example of why a security layer between the GCS/companion computer and the autopilot is needed for any deployment using autonomous agents or AI co-pilots.

We built a MAVLink bridge in SINT Protocol that addresses exactly this — not as a replacement for ArduPilot's internal sanitization, but as a **defense-in-depth layer upstream** that prevents unsanitized commands from reaching the flight controller in the first place.

**How it works:** The `MAVLinkInterceptor` sits between the source (GCS, companion computer, AI agent) and the autopilot serial/UDP port. Every `SET_POSITION_TARGET_LOCAL_NED`, `COMMAND_LONG`, and `COMMAND_INT` is routed through a `PolicyGateway` that:

1. Validates a cryptographic capability token (Ed25519, W3C DID did:key)
2. Checks physical constraints (`maxVelocityMps`, `geofence`, `timeWindow`)
3. Assigns an approval tier — `SET_POSITION_TARGET` is T2_ACT (requires escalation), not auto-allowed
4. Logs every decision to a SHA-256 hash-chained evidence ledger

For the specific case in this issue (`SET_POSITION_TARGET_GLOBAL_INT` with extreme altitude), the token constraint check would catch `altitude = 99999m` before it reaches ArduPilot — and the hash chain provides a tamper-evident audit trail.

Key tier assignments:
- `MAV_CMD_COMPONENT_ARM_DISARM` → **T3_COMMIT** (explicit human approval required)
- `MAV_CMD_MISSION_START` → **T3_COMMIT**
- `SET_POSITION_TARGET_LOCAL_NED` → **T2_ACT** (review, not auto-allow)
- `MAV_CMD_DO_FENCE_ENABLE` (disable) → **T3_COMMIT** (removing safety boundary)
- Camera commands → **T0_OBSERVE** (forwarded immediately)

Implementation: [`packages/bridge-mavlink`](https://github.com/pshkv/sint-protocol/tree/master/packages/bridge-mavlink) — 15 tests, zero false positives on camera commands.

Happy to discuss integration patterns for ArduPilot companion computer setups (Raspberry Pi, Jetson, Orange Cube companion).
